src/pages/api/update-training-bookedspot.ts

```diff
@@ -0,0 +1,34 @@
+import { withApiAuthRequired } from '@auth0/nextjs-auth0';
+import {
+	getTrainingWithId,
+	updateTrainingWithId,
+} from '@data/trainingOrderSupabase';
+import { TrainingSupbaseDataType } from '@data/types/trainingTypes/TypeOrthoexTrainingData';
+
+export default withApiAuthRequired(async function updateTrainingBookspot(
+	req,
+	res,
+) {
+	try {
+		const { id } = req.body as {
+			id: number;
+		};
+		let updatedData: TrainingSupbaseDataType | null = null;
+		const trainingData = await getTrainingWithId(id);
+		if (trainingData) {
+			updatedData = await updateTrainingWithId(id, {
```

Comment:
To save on DB access, just update straight. if the row doesn't exist the error message will let you know. We just need to give the client a generic message like update failed, you can log the error in the console for troubleshooting purposes.

Reply:
So what i did here was the following;

- getTrainingwithId and UpdateTrainingWithId returns data if true and throws an error with the postgresError.message.
- I then updated directly, so if any of the functions throws an error, it will then enter the catch block and return code 500 with their respective error